docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,361 @@
+Configure the coordinator
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+On the coordinator, add the following lines to the :ref:`config properties file
+<config_properties>` to enable HTTPS support for the server.
+
+.. note::
+
+  Legacy ``keystore`` and ``truststore`` wording is used in property names, even
+  when directly using PEM-encoded certificates. PEM certs generally do not
+  require truststore properties, because this format incorporates its chain of
+  authorizing certificates back to the originating CA (or possibly back to an
+  intermediate CA assigned by a large international CA).
+
+.. code-block:: text
+
+  http-server.https.enabled=true
+  http-server.https.port=8443
+  http-server.https.keystore.path=etc/clustercoord.pem
+
+Possible alternatives for the third line include:
+
+.. code-block:: text
+
+  http-server.https.keystore.path=etc/clustercoord.jks
+  http-server.https.keystore.path=/usr/local/certs/clustercoord.p12
+
+Relative paths are relative to the Trino server's root directory. In a
+``tar.gz`` installation, the root directory is one level above ``etc``.
+
+JKS keystores always require a password, while PEM format certificates can
+optionally require a password. For these cases, add the following line to
+specify the password.
+
+.. code-block:: text
+
+  http-server.https.keystore.key=<keystore-password>
+
+It is possible for a key inside a keystore to have its own password,
+independent of the keystore's password. In this case, specify the key's password
+with the following property:
+
+.. code-block:: text
+
+  http-server.https.keymanager.password=<key-password>
+
+When your Trino coordinator has an authenticator enabled along with HTTPS
+enabled, it disables HTTP access to the :doc:`Web UI </admin/web-interface>`.
+Although not recommended, you can re-enable it by setting:
+
+.. code-block:: text
+
+  http-server.authentication.allow-insecure-over-http=true
+
+When configured to provide HTTPS connections as shown above, the server
+continues to allow HTTP connections to all clients except the Web UI. When
+you are certain HTTPS connections are stable and reliable from the clients of
+interest, you can disable HTTP access:
+
+.. code-block:: text
+
+  http-server.http.enabled=false
+
+However, there are configuration scenarios that require HTTP to be enabled, even
+though insecure, at the same time HTTPS is enabled for client access. In such
+cases, configure both server types as ``enabled=true``:
+
+.. code-block:: text
+
+  http-server.http.enabled=true
+  http-server.https.enabled=true
+
+Test configuration
+^^^^^^^^^^^^^^^^^^
+
+To test your configuration settings, restart the server and try to connect to it
+with the Trino :doc:`CLI </installation/cli>` or :doc:`Web UI
+</admin/web-interface>`. using a URL that begins with ``https://``.
+
+.. _self_signed_limits:
+
+Limitations of self-signed certificates
+---------------------------------------
+
+It is possible to generate a self-signed certificate with the ``openssl``,
```

Comment:
For this section, I would focus the traditional meaning of "self-signed certificates", where an admin creates a new certificate for a server that is not trusted by anyone.  This is done for expediency as it doesn't require getting trust signoff from anyone else.  So I would drop the paragraphs that talk about CSRs, signing, and internal CAs, as all of these are techniques for interacting with a Global or Organizational Certificate Authority.

I would say the important point is that the using certificates like this require distributing the truststore to every client and configuring every client.  It is also difficult to work with servers like this as browsers don't work well with the server.

Reply:
that section is now rewritten